pkg/controller/data_download_controller.go

```diff
@@ -163,6 +163,11 @@ func (r *DataDownloadReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 		log.Info("Restore PVC is ready")
 
+		if r.dataPathMgr.IsConcurrentExceed() {
```

Comment:
This doesn't fix the problem all the time:

Say current concurrent num is max-1, so only one more is allowed
Then two calls r.dataPathMgr.IsConcurrentExceed at the same time, they both get permitted since concurrent num is not changed for each call
Then they both call dataPathMgr.CreateFileSystemBR and finally one get passed and the other one is errorOut
A better way is we move the call to r.dataPathMgr.CreateFileSystemBR to here. Don't need to add IsConcurrentExceed function